dandi/download.py

```diff
@@ -1035,3 +1032,15 @@ def feed(self, path: str, status: dict) -> Iterator[dict]:
             lgr.warning(
                 "Unexpected download status dict for %r received: %r", path, status
             )
+
+
+def multiasset_target(url_path: str, asset_path: str) -> str:
+    """
+    When downloading assets for a `MultiAssetURL` with `~MultiAssetURL.path`
+    equal to ``url_path``, calculate the path (relative to the output path) at
+    which to save the asset with path ``asset_path``.
+    """
+    prefix = op.dirname(url_path.rstrip("/"))
```

Comment:
I think I might have `lstrip('/')`-ed both url_path and asset_path for consistency here since upstairs asset_path is lstripped but not sure if url_path is, which could then bring odd results below in `asset_path[len(prefix) :]`... and then have one test case which relies on ensuring that they are lstripped

Reply:
It's my understanding that Dandi Archive doesn't permit asset paths with leading slashes, and thus a `path` argument in a URL with a leading slash would match nothing.  Is that not the case?

Reply:
It turns out the archive does allow paths with leading slashes; filed https://github.com/dandi/dandi-archive/issues/1111.  In the meantime, slashes are now stripped from both ends of `url_path`.